### PR TITLE
doc-gen(mergeDecoratorDocs): capture all the metadata docs

### PR DESCRIPTION
**Pull request by petebacondarwin**
Iterate through the docs and merge all the of metadata docs
for each decorator doc into the decorator doc and remove it
so that it is not rendered in a page of its own.

This is a first step in tidying up the decorator API docs.
With this change you will no longer see the "auxiliary" classes, such as
`ComponentMetadata`, `ComponentType` and so on in the API docs.

Instead these docs are now attached to the decorator doc, which can then
be accessed in templates for rendering the description and so on.

### CODE REVIEW

**Comment by petebacondarwin** [comment]
CC @IgorMinar

**Comment by IgorMinar** [comment]
looks good to me. now we need to adjust the templates to render the mergerged docs. anything else?

**Comment by petebacondarwin** [comment]
Let's see if how far we can get by updating the templates. Obviously there are still issues around NgModule, etc

**Comment by IgorMinar** [comment]
the pipe commit looks good.

File: tools/api-builder/docs-package/processors/mergeDecoratorDocs.js [hunk 1]
```diff
@@ -21,7 +21,11 @@ module.exports = function mergeDecoratorDocs() {
         var makeDecorator = getMakeDecoratorCall(doc);
         if (makeDecorator) {
           doc.docType = 'decorator';
+          // get the type of the decorator metadata
           doc.decoratorType = makeDecorator.arguments[0].text;
+          // clear the symbol type named (e.g. ComponentMetadataFactory) since it is not needed
+          console.log(doc.symbolTypeName);
```
Comment (petebacondarwin) [inline tools/api-builder/docs-package/processors/mergeDecoratorDocs.js]: oops that needs to go

**Comment by IgorMinar** [comment]
landed as  13878efca75a5c45f07ca06d42f8e030dd6693c4

**Pull request closed without merging**